resources/deployments/model/deployments.go

```diff
@@ -194,18 +302,26 @@ func (d *DeploymentsModel) GetDeploymentForDeviceWithCurrent(ctx context.Context
 		return nil, nil
 	}
 
-	link, err := d.imageLinker.GetRequest(ctx, deployment.Image.Id,
+	if err := d.assignArtifact(ctx, deployment, deviceDeployment, installed); err != nil {
```

Comment:
We don't forbid it on API so it could happen, mainly depends on client implementation. Doe in the example you are mentioning I would not imagine why it would change.